File: Strings/Easy/345. Reverse Vowels of a String.py
'''Given a string s, reverse only all the vowels in the string and return it.

The vowels are 'a', 'e', 'i', 'o', and 'u', and they can appear in both cases.

 

Example 1:

Input: s = "hello"
Output: "holle"
Example 2:

Input: s = "leetcode"
Output: "leotcede"'''
\

class Solution:
    def reverseVowels(self, s) -> str:
        vowels=['a','e','i','o','u','A','E','I','O','U']
        s=list(s)
        left=0;
        right=len(s)-1
        while left<right:
            left_char=s[left]
            right_char=s[right]
            if left_char in vowels and right_char in vowels:
                s[left],s[right]=s[right],s[left]
                left+=1
                right-=1
            elif left_char in vowels:
                right-=1
            elif right_char in vowels:
                left+=1
            else:
                left+=1
                right-=1

        return ''.join(s)

test=Solution()
print(test.reverseVowels("hello"))
        
        

File: Strings/Easy/345. Reverse Vowels of a String.js
`Given a string s, reverse only all the vowels in the string and return it.

The vowels are 'a', 'e', 'i', 'o', and 'u', and they can appear in both lower and upper cases, more than once.

 

Example 1:

Input: s = "hello"
Output: "holle"
Example 2:

Input: s = "leetcode"
Output: "leotcede"
 `;

var reverseVowels = function (s) {
  let left = 0;
  let right = s.length - 1;
  const vowels = "aeiouAEIOU";
  s = s.split("");
  while (left < right) {
    const curr = s[left];
    const last = s[right];
    console.log(curr, last);
    if (vowels.includes(curr) && vowels.includes(last)) {
      //! swap the letters
      [s[left], s[right]] = [s[right], s[left]];
      left++;
      right--;
    } else if (vowels.includes(curr)) {
      right--;
    } else if (vowels.includes(last)) {
      left++;
    } else {
      left++;
      right--;
    }
  }
  return s.join("");
};

console.log(reverseVowels("leetcode"));
